### PR TITLE
WIP shell interactive

### PR DESCRIPTION
**Pull request by pana1990**
- [x] Add phpdocs

- [x] Update readme (description, installation, usage...)

- [ ] check code

- [x] Register in packagist

next : 

- [ ] Add extesion in app-basic (require-dev)

- [ ] Add extesion in app-advanced (require-dev)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -11,11 +14,23 @@
...
+        {
+            "name": "Daniel Gomez Pan",
+            "email": "pana_1990@hotmail.com"
+        }
+    ],
     "require": {
         "yiisoft/yii2": "~2.0.0",
-        "phpunit/phpunit": "~4.4"
+        "phpunit/phpunit": "~4.4",
+        "psy/psysh": "0.7.*"
```
Comment (LAV45) [inline composer.json]: Move 
```
        "phpunit/phpunit": "~4.4",
        "psy/psysh": "0.7.*"
```
to "require-dev"

Comment (pana1990) [inline composer.json]: why? 

if you use this extension into basic or advanced template you can include require-dev.

Howover, i think to remove phpunit, it is not using.

Comment (cebe) [inline composer.json]: just remove phpunit entirely and leave the rest as is. phpunit is provided by the ci environment if needed.

Comment (pana1990) [inline composer.json]: done! :+1:

File: README.md [hunk 2]
```diff
@@ -1,3 +1,38 @@
-# Interactive shell extension for Yii2 framework
+Interactive Shell Extension for Yii 2
+=====================================
+
+This extension provides interactive shell for [Yii framework 2.0](http://www.yiiframework.com).
```
Comment (cebe) [inline README.md]: I think this should mention `psysh` directly. e.g. 

> `This extension provides an interactive shell for [Yii framework 2.0](http://www.yiiframework.com) based on [psysh](http://psysh.org/).`

Comment (pana1990) [inline README.md]: done!

**Comment by pana1990** [comment]
@cebe @samdark @SilverFire @klimov-paul  can you please register this repo in packagist?

**Comment by samdark** [comment]
Done.

**Comment by pana1990** [comment]
thank you very much

File: ShellCaster.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+use Symfony\Component\VarDumper\Caster\Caster;
+use yii\db\ActiveRecord;
+
+/**
+ * ShellCaster provides wrapper for casters of psysh
+ *
+ * @author Daniel Gomez Pan <pana_1990@hotmail.com>
+ * @since 2.0
+ */
+class ShellCaster
```
Comment (samdark) [inline ShellCaster.php]: Probably YiiCaster is more precise name since it works with Yii-specific objects.

Comment (pana1990) [inline ShellCaster.php]: yeah, done!

**Comment by samdark** [comment]
Looks OK to me. @yiisoft/core-developers any additional comments before merge?

**Comment by cebe** [comment]
merge it, we can clean it up later if anything comes up.

**Pull request merged**